public/components/layer_control_panel/hide_layer_button.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiButtonIcon, EuiFlexItem } from '@elastic/eui';
+import React, { useState } from 'react';
+import { Map as Maplibre } from 'maplibre-gl';
+import {
+  LAYER_PANEL_HIDE_LAYER_ICON,
+  LAYER_PANEL_SHOW_LAYER_ICON,
+  LAYER_VISIBILITY,
+} from '../../../common';
+import { MapLayerSpecification } from '../../model/mapLayerType';
+import { updateLayerVisibility } from '../../model/map/layer_operations';
+
+interface MaplibreRef {
+  current: Maplibre | null;
+}
+
+interface HideLayerProps {
+  layer: MapLayerSpecification;
+  maplibreRef: MaplibreRef;
+  updateLayer: (layer: MapLayerSpecification) => void;
+}
+export const HideLayer = ({ layer, maplibreRef, updateLayer }: HideLayerProps) => {
+  const [layerVisibility, setLayerVisibility] = useState(
+    new Map<string, boolean>([[layer.id, layer.visibility === LAYER_VISIBILITY.VISIBLE]])
+  );
+
+  const onLayerVisibilityChange = () => {
+    let updatedLayer: MapLayerSpecification;
```

Comment:
create a new object with only visibility instead of entire layer.

Reply:
We also need to update the existing layer, so we need the layer object to replace the existing:
```
updatedLayer = {
        ...layer,
        visibility: LAYER_VISIBILITY.NONE,
      };
updateLayer(updatedLayer);
```

Reply:
Updated to have a function `updateLayerVisibility` to only pass the visibility props